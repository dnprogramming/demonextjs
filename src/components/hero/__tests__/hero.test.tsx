import { render, screen } from "@testing-library/react";
import Hero from "../hero";
import React from "react";
import HomeContent from "@/content/home";

describe("Hero Component", () => {
    it("renders hero header", () => {
        render(<Hero hero={HomeContent.hero!} />);
        const ctaHeading = screen.getByText(HomeContent.hero!.ctaHeading);
        const ctaDescription = screen.getByText(HomeContent.hero!.ctaDescription);

        expect(ctaHeading).toBeInTheDocument();
        expect(ctaDescription).toBeInTheDocument();
    });

    it("renders hero component unchanged", () => {
        const { container } = render(<Hero hero={HomeContent.hero!}/>);
        expect(container).toMatchSnapshot();
    });
});