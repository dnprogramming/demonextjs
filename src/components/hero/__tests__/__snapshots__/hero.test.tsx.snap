// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Hero Component renders hero component unchanged 1`] = `
<div>
  <img
    alt="Demo Hero Banner"
    class="heroImage"
    data-nimg="1"
    decoding="async"
    height="1080"
    loading="eager"
    src="/_next/image?url=%2Fimages%2Fhero_banner.png&w=3840&q=75"
    srcset="/_next/image?url=%2Fimages%2Fhero_banner.png&w=1920&q=75 1x, /_next/image?url=%2Fimages%2Fhero_banner.png&w=3840&q=75 2x"
    style="color: transparent;"
    width="1920"
  />
  <div
    id="hero"
  >
    <h1>
      Need an application developed?
    </h1>
    An applications can take time and resources to develop but the return on investment can be exponential by having it built by a quality team.
  </div>
</div>
`;
